packages/sdk-codegen/src/sdkModels.spec.ts

```diff
@@ -71,38 +73,62 @@ describe('sdkModels', () => {
   })
 
   describe('camelCase', () => {
-    it('empty is empty', () => {
-      expect(camelCase('')).toEqual('')
-    })
-    it('foo-bar is fooBar', () => {
-      expect(camelCase('foo-bar')).toEqual('fooBar')
-    })
-    it('foo_bar is fooBar', () => {
-      expect(camelCase('foo_bar')).toEqual('fooBar')
-    })
-    it('foobar is foobar', () => {
-      expect(camelCase('foobar')).toEqual('foobar')
+    it.each<[string, string]>([
+      ['', ''],
+      ['foo-bar', 'fooBar'],
+      ['foo -bar', 'fooBar'],
+      ['foo- bar', 'fooBar'],
+      ['foo --  bar', 'fooBar'],
+      ['foo bar', 'fooBar'],
+      ['foo bar--', 'fooBar'],
+      ['foo bar   ', 'fooBar'],
+      ['foo_bar', 'fooBar'],
+      ['foo   bar', 'fooBar'],
+      ['foo -  bar  - - baz', 'fooBarBaz'],
+      ['Foo  -  Bar', 'FooBar'],
+      ['FOOBAR', 'FOOBAR'],
+    ])('"%s" is "%s"', (actual, expected) => {
+      expect(camelCase(actual)).toEqual(expected)
     })
-    it('FOOBAR is FOOBAR', () => {
-      expect(camelCase('FOOBAR')).toEqual('FOOBAR')
+  })
+
+  describe('safeName', () => {
+    it.each<[string, string]>([
+      ['', ''],
+      ['foo-bar', 'foo_bar'],
```

Comment:
do we only need to worry about kebab and space names?

Reply:
To answer your top-level question: I'm going to look into the possibility of aliasing the json keys for the Typescript information. However, I think that's going to require completely changing the way we do the types.

Regarding conversion to `safeName()` .. as I mentioned in `TODO` for that function, there's no unicode recognition or processing right now. Our near-term need is to support hyphens. I just thought we should support spaces as well because someone might try to use that.

I think any special symbol names in an API spec is a bad idea when so many programming languages will have issues with them, so the current implementation is a bit of a forcing function that will cause the code generator to break when symbol names start to get too "out there." This will prompt a conversation with the implementer of the questionable endpoint implementation.